model/project_parser.go

```diff
@@ -397,45 +397,52 @@ func (pss *parserStringSlice) UnmarshalYAML(unmarshal func(interface{}) error) e
 	return nil
 }
 
-// LoadProjectInto loads the raw data from the config file into project
-// and sets the project's identifier field to identifier. Tags are evaluateed.
-func LoadProjectInto(data []byte, identifier string, project *Project) error {
-	p, errs := projectFromYAML(data)
-	if len(errs) > 0 {
-		// create a human-readable error list
-		buf := bytes.Buffer{}
-		for _, e := range errs {
-			if len(errs) > 1 {
-				buf.WriteString("\n\t") //only newline if we have multiple errs
-			}
-			buf.WriteString(e.Error())
+func LoadProjectFromVersion(v *Version, identifier string, shouldSave bool) (*Project, error) {
+	if v.ParserProject != nil {
+		pp, errs := translateProject(v.ParserProject)
+		return pp, formatErrors(errs)
+	}
+	if v.Config == "" {
+		return nil, errors.New("version has no config")
+	}
+	p := &Project{}
+	pp, err := LoadProjectInto([]byte(v.Config), identifier, p)
+	if err != nil {
+		return nil, errors.Wrap(err, "error loading project")
+	}
+	if shouldSave {
+		if err := UpdateVersionProject(v.Id, pp); err != nil {
+			return nil, errors.Wrap(err, "error updating version with project")
 		}
-		return errors.Errorf("%s: %s", LoadProjectError, buf.String())
 	}
-	*project = *p
-	project.Identifier = identifier
-	return nil
+	v.ParserProject = pp
+	return p, nil
 }
 
-// projectFromYAML reads and evaluates project YAML, returning a project and warnings and
-// errors encountered during parsing or evaluation.
-func projectFromYAML(yml []byte) (*Project, []error) {
-	intermediateProject, errs := createIntermediateProject(yml)
-	if len(errs) > 0 {
-		return nil, errs
+// LoadProjectInto loads the raw data from the config file into project
+// and sets the project's identifier field to identifier. Tags are evaluated.
+// Returns the intermediate step.
+func LoadProjectInto(data []byte, identifier string, project *Project) (*ParserProject, error) {
+	intermediateProject, err := createIntermediateProject(data)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating parser project")
 	}
+
+	// return project even with errors
 	p, errs := translateProject(intermediateProject)
-	return p, errs
+	*project = *p
+	project.Identifier = identifier
+	return intermediateProject, errors.Wrap(formatErrors(errs), "error translating project")
 }
 
 // createIntermediateProject marshals the supplied YAML into our
 // intermediate project representation (i.e. before selectors or
 // matrix logic has been evaluated).
-func createIntermediateProject(yml []byte) (*parserProject, []error) {
-	p := &parserProject{}
+func createIntermediateProject(yml []byte) (*ParserProject, error) {
+	p := &ParserProject{}
 	err := yaml.Unmarshal(yml, p)
 	if err != nil {
-		return nil, []error{err}
+		return nil, formatErrors([]error{err})
```

Comment:
`return nil, err` (I don't think `formatErrors([]error{err})` does anything except put the error in a slice and then take it out again.)